balance/adjustment.py

```diff
@@ -179,14 +185,45 @@ def trim_weights(
                 logger.debug("No extreme weights were trimmed")
     elif weight_trimming_percentile is not None:
         # Winsorize
+        percentile = weight_trimming_percentile
+        if isinstance(percentile, (list, tuple, np.ndarray)):
+            if len(percentile) != 2:
+                raise ValueError(
+                    "weight_trimming_percentile must be a single value or a length-2 iterable"
+                )
+            lower_limit, upper_limit = percentile
+        else:
+            lower_limit = upper_limit = percentile
+
+        def _validate_limit(limit: Union[float, int, None]) -> Union[float, None]:
```

Comment:
Probably worth moving this outside, and adding documentation and testing.
I'll move to accept the diff as is - just flagging it's probably better that way.